src/service/metrics/otlp.rs

```diff
@@ -761,18 +791,19 @@ fn process_hist_data_point(
 
     // add bucket records
     let len = data_point.bucket_counts.len();
+    let mut accumulated_count = 0;
     for i in 0..len {
         let mut bucket_rec = rec.clone();
         bucket_rec[NAME_LABEL] = format!("{}_bucket", rec[NAME_LABEL].as_str().unwrap()).into();
-        if let Some(val) = data_point.bucket_counts.get(i) {
-            bucket_rec[VALUE_LABEL] = (*val as f64).into()
-        }
         if let Some(val) = data_point.explicit_bounds.get(i) {
             bucket_rec["le"] = (*val.to_string()).into()
-        }
-        if i == len - 1 {
+        } else {
             bucket_rec["le"] = f64::INFINITY.to_string().into();
         }
```

Comment:
Just for my understanding : originally we wanted only the last bucket label to be set as Inf, right? And the issue was if there was no explicit bound, `le` would not be set at all. Now we are assuming that no explicit bound means we are at the last bucket, and hence it should be Inf, instead of using loop counter. Is that correct?

Reply:
The data must follow this structure.